### DISCUSSION TITLE
Chrome v122 password warning pop-up causing issues

### DISCUSSION THREAD

**Discussion by ccea-dev-team** [selenide/selenide#2658] [Q&A] [answered]
## The problem

Hi there,

See screenshot - this started happening in Chrome v122 and is knocking off some tests.  Is there an option to disable this password data breach warning?

## Tell us about your environment
* **Selenide Version**:  7.1.0
* **Chrome\Firefox\IE Version**:  Chrome v122

![Capture](https://github.com/selenide/selenide/assets/99395943/84033da6-3762-45af-b186-4543de18d09a)

**Comment by asolntsev** [comment, marked as answer]
@ccea-dev-team I cannot reproduce the problem. 
Could you please try to add these options to Chrome before start?

```java
preferences.put("credentials_enable_service", false);
preferences.put("profile.password_manager_enabled", false);
```

if it helps, we can add these options to Selenide by default.

**Reply by ccea-dev-team** [reply]
Hi,

Thanks for the reply.  Added those prefs (seems they are experimental options) and it's working now -

    ChromeOptions options = new ChromeOptions();
    options.addArguments("--start-maximized");
    Map<String, Object> prefs = new HashMap<String, Object>();
    prefs.put("credentials_enable_service", false);
    prefs.put("password_manager_enabled", false); 
    options.setExperimentalOption("prefs", prefs);
    Configuration.browserCapabilities = options;

**Reply by johnhaavardstun** [reply]
We also ran into this issue - after trying a few things (this included), we're currently using
```java
System.setProperty("chromeoptions.prefs", "profile.password_manager_leak_detection=false");
```

We couldn't set `prefs` using `Configuration.browserCapabilities` because those completely override Selenide's defaults, including the way downloads directory is set.

(If you're managing your driver to a greater extent than we do, then the above solution is probably fine.)

**Reply by BorisOsipov** [reply]
>We couldn't set prefs using Configuration.browserCapabilities because those completely override Selenide's defaults, including the way downloads directory is set.

Can you create the issue with repro? it is not desired behaviour, prefs should be merged.

**Reply by johnhaavardstun** [reply]
Seems to be an issue (or deliberate choice, I dunno) in Selenium - the `experimentalOptions` are merged, but only shallowly:

https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/chromium/ChromiumOptions.java#L335

Selenide could probably work around that by merging `experimentalOptions.prefs` itself before merging the capabilities, if that's desired.

I don't have the time today to create an actual repro, but I'll look into it during the week if you want me to.

**Comment by asolntsev** [comment]
Added the needed option in PR https://github.com/selenide/selenide/pull/2662 (Selenide 7.2.0)

**Reply by Redbeeard** [reply]
@asolntsev Hello! I can see your change in version 7.2.0 but we still get this problem in our tests. Does this problem still persist?

**Reply by asolntsev** [reply]
I am not sure, probably it's a bug in Chrome that needs a fix on Chrome side.

**Comment by ccea-dev-team** [comment]
Seems ChromeDriver for v122 has broken a few things related to popups and file downloads.  We are now experience failing tests due to file downloads being blocked as insecure.  Issues have been raised in the ChromeDriver bug reporting site - 

![image](https://github.com/selenide/selenide/assets/99395943/fd3e2dbc-b33d-4219-b593-5235556a47c2)

![image](https://github.com/selenide/selenide/assets/99395943/56361c56-d6c6-4811-bb15-0b0e2ef23552)

**Comment by ccea-dev-team** [comment]
@asolntsev 

Hi, another chrome v122 issue.  Do you know what settings can be used to disable the "insecure download" message as outlined here - https://bugs.chromium.org/p/chromedriver/issues/detail?id=4730 

This is causing any tests we have that tries to download files to fail.  Can't find any options online to disable it.

![down](https://github.com/selenide/selenide/assets/99395943/aa779a2c-0ad7-4861-9eb9-b8b4eaf81d1d)

**Reply by asolntsev** [reply]
@ccea-dev-team  I think Selenide already adds the needed options. 
I guess it's probably a bug in Chrome 122. Just need to wait for the fix.

**Comment by ruwanrcs** [comment]


**Reply by asolntsev** [reply]
@ruwanrcs What Selenide version do you use? I think this problem ("it resets the default file download path") was fixed in Selenide 7.4.0

**Reply by ruwanrcs** [reply]
@asolntsev Thank you for the suggestion regarding Selenide version 7.4.0. However, our current framework does not use Selenide, so this issue ("it resets the default file download path") is unrelated to Selenide. We're using native Selenium to configure browser capabilities, specifically ChromeOptions, and handling browser configuration through an external properties file. Unfortunately, due to the constraints of our project, we are unable to add new libraries, including Selenide. Therefore, we are looking for a solution within the existing Selenium-based setup.

**Reply by asolntsev** [reply]
@ruwanrcs wtf... Then why you submitted your question to Selenide github?

**Reply by ruwanrcs** [reply]
Sorry for the mix up.. We’re actually looking to add Selenide, but it requires approval. In the meantime, I was hoping to find any adjustments we could make in our current code to bypass the file download path reset issue.

**Reply by asolntsev** [reply]
Then what exactly you are asking for? Just use the latest Selenide 7.5.0+, and everything will work fine.
